Update health checks docs for tproxy (#10419)
hashicorp/consul: website/content/docs/k8s/connect/health.mdx

# Kubernetes Health Checks in Consul on Kubernetes

~> This topic requires familiarity with [Kubernetes Health Checks](https://kubernetes.io/docs/tasks/configure-pod-container/configure-liveness-readiness-startup-probes/).

> This page describes how Consul on Kubernetes will sync the Kubernetes health probes of a pod to Consul for service mesh use cases.

Reviewer: ```suggestion
This page describes how Consul on Kubernetes will sync the status of Kubernetes health probes of a pod to Consul for service mesh use cases.
```

Health check synchronization with Consul is done automatically whenever `connectInject.enabled` is `true`.

For each Kubernetes pod that is connect-injected the following will be configured:

1. A [TTL health check](/docs/discovery/checks#ttl) is registered within Consul.
The Consul health check's state will reflect the pod's readiness status,
which is the combination of all Kubernetes probes registered with the pod.

> 1. If the pod is utilizing [Transparent Proxy](/docs/connect/transparent-proxy) mode the mutating webhook will also mutate all `http` based Startup, Liveness, and Readiness probes in the pod to redirect through the Envoy proxy.

Reviewer: ```suggestion
1. If the pod is utilizing [Transparent Proxy](/docs/connect/transparent-proxy) mode, the mutating webhook will mutate all `http` based Startup, Liveness, and Readiness probes in the pod to redirect through the Envoy proxy.
```

> Consul will also use an `ExposePath` exception for each probe so that kubelet can access the endpoint through the Envoy proxy.

Reviewer: ```suggestion
This is done with [`ExposePaths` configuration](https://www.consul.io/docs/connect/registration/service-registration#expose-paths-configuration-reference) for each probe so that kubelet can access the endpoint through the Envoy proxy.
```

Reviewer 2: good suggestion, just use a relative link `docs/connect/registration/service-registration#expose-paths-configuration-reference`

Reviewer: Ooops! good catch, I've updated my suggestion.

> ~> The mutation behaviour can be disabled via setting the annotation `consul.hashicorp.com/transparent-proxy-overwrite-probes` to `false`.

Reviewer: ```suggestion
~> The mutation behaviour can be disabled by either setting the `consul.hashicorp.com/transparent-proxy-overwrite-probes` pod annotation to `false` or the `connectInject.defaultOverwriteProbes` Helm value to `false`.
```
